### PR TITLE
This is the control system that worked well

### PR DESCRIPTION
**Pull request by mschottdorf**
Featuring a user-adjustable max-flow setting, VTE estimates, and oxygen sensor readings. And removed some of the dead wood

@sneakers-the-rat These should be the final edits to the control system for Mk 1
@zajdel This is the version of the code that you have been using on the pi

### CODE REVIEW

File: vent/io/devices/base.py [hunk 1]
```diff
@@ -477,11 +477,11 @@ def _read_conversion(self, **kwargs) -> int:
             self.write_register(self.pointer.P.pack('CONFIG'), self.cfg)
             self._last_cfg = self.cfg
             data_rate = self._config.DR.unpack(self.cfg)
-            while not (self._ready() or mode == 'CONTINUOUS'):
+            '''while not (self._ready() or mode == 'CONTINUOUS'):
                 # TODO: Needs timout
                 tick = time.time()
                 while (time.time() - tick) < (1 / data_rate):
-                    pass  # TODO: implement asyncio.sleep()
+                    pass  # TODO: implement asyncio.sleep()'''
```
Comment (mschottdorf) [inline vent/io/devices/base.py:484]: this was causing a lot of trouble, and lead to considerable speadup

File: vent/io/devices/sensors.py [hunk 2]
```diff
@@ -284,9 +284,10 @@ def _convert(self, raw) -> float:
         raw = super()._convert(raw)
         fit_param = 2.5837e-05
         if(raw >= 0):
-            converted_flow = (-1.0*np.sqrt(raw)/np.sqrt(fit_param))
+            #converted_flow = (-1.0*np.sqrt(raw)/np.sqrt(fit_param))
+            converted_flow = 192.6426*(raw)**(1/1.9128)
         else:
-            converted_flow = (1.0*np.sqrt(-1.0*raw)/np.sqrt(fit_param))
+            converted_flow = -192.6426*(np.abs(raw))**(1/1.9128)
```
Comment (mschottdorf) [inline vent/io/devices/sensors.py:290]: and here is why we could not calculate VTE for a month #facepalm

File: vent/controller/control_module.py [hunk 3]
```diff
@@ -178,7 +178,7 @@ def _initialize_set_to_COPY(self):
         with self._lock:
         # Copy of the SET variables for threading.
             self.COPY_SET_PIP       = self.__SET_PIP 
-            self.COPY_SET_PIP_TIME  = self.__SET_PIP_TIME
+            self.COPY_SET_PIP_TIME  = self.__SET_PIP_GAIN
```
Comment (mschottdorf) [inline vent/controller/control_module.py:181]: @sneakers-the-rat whatever we want to call this... gain/pip_time/flow... You seem to have a talent for human-machine interactions :-)

File: vent/controller/control_module.py [hunk 4]
```diff
@@ -501,67 +499,6 @@ def __start_new_breathcycle(self):
             self.dl.flush_logfile()        # If we kept records, flush the data from the previous breath cycle
             self.dl.rotation_newfile()     # And Check whether we run out of space for the logger
 
-    def _STATECONTROL_update(self, dt):
```
Comment (mschottdorf) [inline vent/controller/control_module.py:504]: this stuff is all gone. As per our discussion: we only support and test the PID system

File: vent/io/devices/sensors.py (hunk 2 above)
Comment (zajdel) [reply vent/io/devices/sensors.py:290]: Yeah, it was only 2 weeks, actually! This is Sophie's calibration, replacing the old calibration file.

File: vent/controller/control_module.py (hunk 3 above)
Comment (zajdel) [reply vent/controller/control_module.py:181]: We call it "flow" on the GUI, and it ranges from 1-5. Tuning it is fun and just seems to work.

**Comment by zajdel** [comment]
I approve!

**Comment by mschottdorf** [comment]
I'll merge this, so that we have the frozen controller in the main repo.

**Pull request merged**